package.json

```diff
@@ -53,36 +53,28 @@
   },
   "homepage": "https://github.com/akahu-io/akahu-sdk-js#readme",
   "jest": {
-    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
-    "moduleFileExtensions": [
-      "js",
-      "ts"
-    ],
-    "transform": {
-      ".ts": "ts-jest"
-    },
     "collectCoverage": true,
     "coverageDirectory": "coverage"
   },
   "devDependencies": {
+    "@babel/preset-env": "^7.24.5",
+    "@babel/preset-typescript": "^7.24.1",
     "@rollup/plugin-typescript": "^8.2.3",
-    "@types/jest": "^26.0.24",
-    "@types/node": "^16.4.12",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^18.19.33",
     "@types/uuid": "^8.3.1",
-    "axios-mock-adapter": "^1.19.0",
-    "jest": "^27.0.6",
+    "axios-mock-adapter": "^1.22.0",
+    "jest": "^29.7.0",
     "prettier": "^2.8.8",
     "rollup": "^2.55.0",
-    "ts-jest": "^27.0.4",
-    "typedoc": "^0.21.5",
-    "typedoc-plugin-markdown": "^3.10.4",
-    "typedoc-plugin-merge-modules": "^3.0.0",
-    "typedoc-plugin-no-inherit": "^1.3.0",
-    "typescript": "^4.3.5"
+    "typedoc": "^0.25.13",
+    "typedoc-plugin-markdown": "^4.0.1",
+    "typedoc-plugin-merge-modules": "^5.1.0",
+    "typedoc-plugin-no-inherit": "^1.4.0",
+    "typescript": "^5.4.5"
   },
   "dependencies": {
-    "axios": "^0.21.1",
-    "tslib": "^2.3.0",
+    "axios": "^1.6.8",
```

Comment:
Should this be a major version bump for our Package?

Does it have potential to break things such as args passed in, or middlewares?

Reply:
We only expose a fairly limited subset of native `axios` options, the most significant of which is `adapter`, which doesn't appear to have any breaking API changes as far as I can tell. That would indicate that we're maintaining API compatibility, so a minor version bump should be sufficient.

However, we specifically added the `adapter` param to support [axios-fetch-adaptor](https://github.com/haverstack/axios-fetch-adapter) for use in environments like the CloudFlare worker runtime which doesn't implement the necessary APIs for axios to run natively. It looks like this [isn't compatible with axios 1.x](https://github.com/haverstack/axios-fetch-adapter/issues/7), not due to API compatibility, but changes in import paths in the `axios` peer-dependency. Nonetheless, it would mean that the new version does break this use-case, and because we're specifying `axios` as a dependency rather than peer-dependency, I'm not sure whether `npm` would be smart enough to find the right combination of package versions to maintain compatibility.